### PR TITLE
infra: fix regression-report.yml and add a new regression-report.sh

### PR DESCRIPTION
**Pull request by relentless-pursuit**
Adds the new official releases for extractor and diff-java-tool and a new regression-report.sh to keep the operations and changes for diff-report.yml and regression-report.yml as separate for now. 

This PR will test on the following scenarios:
1. InputJavaFile generating projects in yml format (as we have made the default projects format as yml)
2. PR configurations with .properites file and generating the regression-report in .properties file
3. PR configurations with .yml file and generating the regression-report in .yml file
4. Using the existing config bundle commands Github, generate report for <CheckName/all-in-one-examples> using the default projects list
5. Using the existing config bundle commands Github, generate report for <CheckName/all-in-one-examples> using the custom projects list e.g. WhiteSpaceAround/Example3

### CODE REVIEW

**Comment by relentless-pursuit** [comment]
Tesing for the inputJavaFile: https://github.com/checkstyle-gsoc/checkstyle/actions/runs/11254110878

**Comment by relentless-pursuit** [comment]
Testing for generated configs with custom project list (e.g. RequireThis): https://github.com/checkstyle-gsoc/checkstyle/actions/runs/11255188561

File: .ci/regression-report.sh [hunk 1]
```diff
@@ -0,0 +1,136 @@
+#!/bin/bash
```
Comment (romani) [inline .ci/regression-report.sh:1]: copy paste is too big.
image

can we update diff-report.yml to expect different file name ?

**Review by romani — CHANGES_REQUESTED** [review]
items:

File: .ci/regression-report.sh (hunk 1 above)
Comment (romani) [reply .ci/regression-report.sh:1]: let copy from web file as is.

in  bash we will just get fist by extension (there supposed to be only one property file):
```
property_files=(*.properties)
PROPERTY_FILE=$({files[0]})

```

Comment (romani) [reply .ci/regression-report.sh:1]: please create separate PR to update old workflow, I understand that testing is complicated, I think we can merge this without pre-testing, we will test after merge.

**Comment by relentless-pursuit** [comment]
As we move from `diff-report.yml` to `regression-report.yml` in an incremental manner, we saw that the entire structure of `regression-report.yml` was built with projects.properties from documentation to code. And, we have an update to use `projects.yml`. But, since checkstyle continues to use `diff-report.yml` still dependent on `project.properties`, we made changes to support both format as both the workflows use `diff-report.sh`. And, thereofre, we made a refactoring of `project.properites` to `projects.properties`. 

Merged PR: https://github.com/checkstyle/checkstyle/pull/15759
Runner: https://github.com/checkstyle/checkstyle/actions/runs/11262575457/job/31318677710

**Comment by romani** [comment]
>As we move from diff-report.yml to regression-report.yml

Not yet, they will exist together for several months.
I expect no changes in readme.md

File: .ci/regression-report.sh (hunk 1 above)
Comment (romani) [reply .ci/regression-report.sh]: Workaround at https://github.com/checkstyle/checkstyle/pull/15759

Resolved.

I will keep it "unresolved" in GitHub just to let others see reason easier

**Comment by relentless-pursuit** [comment]
Testing for generated configs with default project list (e.g. AbstractClassName):https://github.com/checkstyle-gsoc/checkstyle/actions/runs/11264005371/job/31323162613

**Comment by relentless-pursuit** [comment]
@romani, testing is done.

**Review by romani — APPROVED** [review]
Ok to merge

**Pull request merged**